src/v/cloud_storage/anomalies_detector.cc

```diff
@@ -196,12 +234,16 @@ ss::future<> anomalies_detector::check_manifest(
           previous_seg_meta,
           _result.detected.segment_metadata_anomalies);
         previous_seg_meta = seg_meta;
+
+        _result.last_scrubbed_offset = seg_meta.committed_offset;
```

Comment:
Is the reason for the quirk about non-linearity just from us not having a reverse iterator? I wonder if it's worth adding one, or collecting the segment names and iterating backwards. At least that way it might be easier to reason about ordering across spillover events, though I guess it's not too important given these checks are best effort

Reply:
> Is the reason for the quirk about non-linearity just from us not having a reverse iterator?

Yes. Collecting the segments and iterating backwards felt too memory intensive. Adding a reverse iterator is a bit tricky, but could be done. It simplifies the semantics of `_last_scrubbed_offset` within the anomalies detector, but outside of it, things remain the same.

We have to encapsulate this complexity somewhere: either directly in the anomaly detection code or in a reverse iterator.

Reply:
Hmm looking around it does seem like there's a `prev(const_iter)` method in `segment_meta_cstore`, though it's not exposed in the partition manifest. Can that not do what we need it to here?

I'm just being extra scrutinous here because it's serialized to storage and a bit tricky/surprising to reason about. I'll concede though that it seems like we should be able to change its semantics in the future given all it's doing is validation.